File: src/main/java/ru/xpendence/topjavagraduation/controller/mapper/VoteMapper.java
package ru.xpendence.topjavagraduation.controller.mapper;

import org.springframework.stereotype.Component;
import ru.xpendence.topjavagraduation.controller.model.request.VoteRequest;
import ru.xpendence.topjavagraduation.controller.model.response.VoteResponse;
import ru.xpendence.topjavagraduation.entity.Restaurant;
import ru.xpendence.topjavagraduation.entity.User;
import ru.xpendence.topjavagraduation.entity.Vote;

import java.time.LocalDate;

@Component
public class VoteMapper {

    public Vote toVote(VoteRequest request, Long userId) {
        return new Vote(
                null,
                LocalDate.now(),
                new User(userId, "", ""),
                new Restaurant(request.restaurantId(), null)
        );
    }

    public VoteResponse toResponse(Vote vote) {
        return new VoteResponse(
                vote.getId(),
                vote.getDate(),
                vote.getUser().getId(),
                vote.getRestaurant().getId()
        );
    }
}

File: src/main/kotlin/ru/xpendence/topjavagraduation/repository/mapper/VoteMapper.kt
package ru.xpendence.topjavagraduation.repository.mapper

import ru.xpendence.topjavagraduation.entity.Restaurant
import ru.xpendence.topjavagraduation.entity.Vote
import java.sql.ResultSet

fun ResultSet.toVote(): Vote =
    Vote(
        id = getLong("id"),
        date = getDate("date").toLocalDate(),
        user = getLong("user_id").toUserStub(),
        restaurant = Restaurant(id = getLong("restaurant_id")),
    )

fun ResultSet.toVoteWithDetails(): Vote =
    Vote(
        id = getLong("id"),
        date = getDate("date").toLocalDate(),
        user = toUserFromJoin(),
        restaurant = Restaurant(
            id = getLong("restaurant_id"),
            name = getString("restaurant_name"),
        ),
    )
